assets/src/stories-editor/components/video-block-edit-with-preview.js

```diff
@@ -0,0 +1,246 @@
+		setAttributes( { poster: image.url } );
+	}
+
+	onRemovePoster() {
+		const { setAttributes } = this.props;
+		setAttributes( { poster: '' } );
+
+		// Move focus back to the Media Upload button.
+		this.posterImageButton.current.focus();
+	}
+
+	onUploadError( message ) {
+		const { noticeOperations } = this.props;
+		noticeOperations.removeAllNotices();
+		noticeOperations.createErrorNotice( message );
+	}
+
+	render() {
+		const {
+			caption,
+			controls,
+			poster,
+			src,
+		} = this.props.attributes;
+		const {
+			instanceId,
+			isSelected,
+			setAttributes,
+		} = this.props;
+		const switchToEditing = () => {
+			this.setState( { editing: true } );
+		};
+		const videoPosterDescription = `video-block__poster-image-description-${ instanceId }`;
+
+		return (
+			<>
+				<BlockControls>
+					<Toolbar>
+						<IconButton
+							className="components-icon-button components-toolbar__control"
+							label={ __( 'Edit video' ) }
+							onClick={ switchToEditing }
+							icon="edit"
+						/>
+					</Toolbar>
+				</BlockControls>
+				<InspectorControls>
+					<PanelBody title={ __( 'Video Settings' ) }>
+						<ToggleControl
+							label={ __( 'Playback Controls' ) }
+							onChange={ this.toggleAttribute( 'controls' ) }
+							checked={ controls }
+						/>
+						<MediaUploadCheck>
+							<BaseControl
+								className="editor-video-poster-control"
+							>
+								<BaseControl.VisualLabel>
+									{ __( 'Poster Image' ) }
+								</BaseControl.VisualLabel>
+								<MediaUpload
+									title={ __( 'Select Poster Image' ) }
+									onSelect={ this.onSelectPoster }
+									allowedTypes={ VIDEO_POSTER_ALLOWED_MEDIA_TYPES }
+									render={ ( { open } ) => (
+										<Button
+											isDefault
+											onClick={ open }
+											ref={ this.posterImageButton }
+											aria-describedby={ videoPosterDescription }
+										>
+											{ ! poster ? __( 'Select Poster Image' ) : __( 'Replace image' ) }
+										</Button>
+									) }
+								/>
+								<p
+									id={ videoPosterDescription }
+									hidden
+								>
+									{ poster ?
+										/* translators: %s: the poster image URL. */
+										sprintf( __( 'The current poster image url is %s' ), this.props.attributes.poster ) :
+										__( 'There is no poster image currently selected' )
+									}
+								</p>
+								{ !! poster &&
+									<Button onClick={ this.onRemovePoster } isLink isDestructive>
+										{ __( 'Remove Poster Image' ) }
+									</Button>
+								}
+							</BaseControl>
+						</MediaUploadCheck>
+					</PanelBody>
+				</InspectorControls>
+				<figure className="wp-block-video">
+					<video
+						autoPlay
+						muted
+						loop
+						controls={ controls }
+						poster={ poster }
+						ref={ this.videoPlayer }
+						src={ src }
+					/>
+					{ ( ! RichText.isEmpty( caption ) || isSelected ) && (
+						<RichText
+							tagName="figcaption"
+							placeholder={ __( 'Write caption…' ) }
```

Comment:
The `__()` calls don't have the AMP textdomain. If it works like PHP, ideally they'll just use the Gutenberg translations.

Reply:
Well, it depends. If a string is ever removed from Gutenberg without us knowing, it won't get translated at all.

It's usually best to include the text domain, even if it means translators have to do a bit more work.

Reply:
Ah, good point.